experiment/run-library.sh

```diff
@@ -572,34 +592,74 @@ function a_out() {
     # Just run the ansible command. No output processing, action logging or measurements
     if $run_lib_dryrun; then
         echo 'FAKE ansible RUN'
-    else
-        ansible $opts_adhoc "$@"
     fi
+
+    ansible $opts_adhoc "$@"
 }
 
 function ap() {
-    local out=$logs/$1; shift
-    mkdir -p "$( dirname $out )"
-    local start="$( date -u +%s )"
-    log "Start 'ansible-playbook $opts_adhoc $*' with log in $out"
     if $run_lib_dryrun; then
         log 'FAKE ansible-playbook RUN'
-        local rc=0
-    else
-        ansible-playbook $opts_adhoc "$@" &>$out && local rc=$? || local rc=$?
+
+        return 0
     fi
+
+    local out=$logs/$1; shift
+    local start="$( date -u +%s )"
+
+    mkdir -p "$logs"
+    log "Start 'ANSIBLE_CALLBACKS_ENABLED='ansible.posix.profile_tasks' ansible-playbook $opts_adhoc $*' with log in $out"
+    ANSIBLE_CALLBACKS_ENABLED='ansible.posix.profile_tasks' ansible-playbook $opts_adhoc "$@" &>$out && local rc=$? || local rc=$?
+
     local end="$( date -u +%s )"
     log "Finish after $(( end - start )) seconds with log in $out and exit code $rc"
 
     measurement_add \
-      "ansible-playbook $opts_adhoc $( _format_opts "$@" )" \
+      "ANSIBLE_CALLBACKS_ENABLED='ansible.posix.profile_tasks' ansible-playbook $opts_adhoc $( _format_opts "$@" )" \
       "$out" \
       "$rc" \
       "$start" \
       "$end" \
       "$katello_rpm" \
       "$satellite_rpm" \
       "$marker"
+
+    return $rc
+}
+
+function apj() {
+    if $run_lib_dryrun; then
+        log 'FAKE ansible-playbook RUN'
+
+        return 0
+    fi
+
+    local test=$1; shift
+    local play_out_json="$logs/$test.json"
+
+    mkdir -p "$logs"
+    log "Start 'ANSIBLE_STDOUT_CALLBACK='ansible.posix.json' ansible-playbook $opts_adhoc $*' with JSON log in $play_out_json"
+    ANSIBLE_STDOUT_CALLBACK='ansible.posix.json' ansible-playbook $opts_adhoc "$@" >$play_out_json && local rc=$? || local rc=$?
+
+    # 'ansible.posix.json' returns datetimes by default ending in 'Z' and without timezone information, so we need to transform it for OPL consumption
+    local play_start_z="$( jq '.plays[].play.duration.start' $play_out_json )"
```

Comment:
If we only care about first play, we should probably use:

```
$ echo '{"plays":[{"play":42},{"play":0}]}' | jq '.plays[0].play'
42
```

Instead of:

```
$ echo '{"plays":[{"play":42},{"play":0}]}' | jq '.plays[].play'
42
0
```

Reply:
Done.